obspy/core/stream.py

```diff
@@ -3118,6 +3119,85 @@ def remove_sensitivity(self, *args, **kwargs):
             tr.remove_sensitivity(*args, **kwargs)
         return self
 
+    def stack(self, group='all', type='linear', npts_tol=0):
+        """
+        Return stream with traces stacked by the same selected metadata.
+
+        :param group: Stack waveforms together which have the same metadata
+            given by this parameter. The parameter should name the
+            corresponding keys of the stats object,
+            e.g. `'{network}.{station}'` for stacking all
+            locations and channels of the stations and returning a stream
+            consisting of one stacked trace for each station.
+            This parameter can take two special values,
+            `'seedid'` which stacks the waveforms by seedid and
+            `'all'` (default) which stacks together all traces in the stream.
+        :param type: Type of stack, one of the following:
+            `'linear'`: average stack (default),
+            `('pw', order)`: phase weighted stack of given order,
+            see [Schimmel1997]_,
+            `('root', order)`: root stack of given order.
+        :param npts_tol: Tolerate traces with different number of points
+            with a difference up to this value. Surplus samples are discarded.
+
+        :returns: New stream object with stacked traces. The metadata of each
```

Comment:
I think returning a new `Stream` instance is the best approach, but would it make sense to perform this operation in place in order to be consistent with most other `Stream` methods?

Reply:
I also think returning a new Stream is the best approach here, because the number of traces changes and a lot of metadata changes, which is not the case for the other, in-place methods.

Reply:
I see two solutions here:

(1) Actually change the existing stream object in-place. This is what a few other method, e.g. `.merge()` already do and it would fit right in with most other stream methods.
(2) Rename the method to `.create_stack()` which would return a new object. Then its IHMO a bit clearer that this does not modify in-place.

I slightly tend towards the first choice but I'm fine with either.

Reply:
I prefer (2) with the name of (1) ;).
But I am also OK with (1) and will change the behavior.